src/main/java/cn/nukkit/entity/Entity.java

```diff
@@ -2273,9 +2273,9 @@ public long getId() {
 
     public void respawnToAll() {
         for (Player player : this.hasSpawned.values()) {
+            this.hasSpawned.remove(player.getLoaderId());
             this.spawnTo(player);
         }
-        this.hasSpawned.clear();
```

Comment:
Why not to move this before the for?

Reply:
> Why not to move this before the for?

Because of this condition in spawnTo()
`public final void spawnTo(Player player) {
        if (!this.hasSpawned.containsKey(player.getLoaderId()) && this.chunk != null && player.usedChunks.containsKey(Level.chunkHash(this.chunk.getX(), this.chunk.getZ()))) {`

Reply:
Ah, anyway make a new copy for it then. Because otherwise that would cause CME imo

Reply:
> Ah, anyway make a new copy of `hasSpawned` for it then. Because otherwise that would cause CME imo

What is CME imo and why can it be caused? I tested my code and no errors occured

Reply:
`ConcurrentModificationException` is caused by concurrent modification so in this case you're removing elements from a collection while iterating over it. Not entirely sure though

Reply:
> `ConcurrentModificationException` is caused by concurrent modification so in this case you're removing elements from a collection while iterating over it. Not entirely sure though

Yeah, you're right. I just fixed my fix

Reply:
I'd rather use ArrayList for that. It should be more suitable for copying